src/MIDebugEngine/MIDebugCommandDispatcher.cs

```diff
@@ -27,7 +27,46 @@ public static Task<string> ExecuteCommand(string command)
 
                 lastProcess = s_processes[s_processes.Count - 1];
             }
-            return ExecuteCommand(command, lastProcess);
+
+            if (lastProcess == null)
+            {
+                throw new InvalidOperationException(MICoreResources.Error_NoMIDebuggerProcess);
+            }
+
+            return lastProcess;
+        }
+
+        public static MICore.ProcessState GetProcessState()
+        {
+            return GetLastProcess().ProcessState;
+        }
+
+        public static async Task<Results> SetSelectedThread(int threadId)
```

Comment:
Can you tell us more about the goal of this method?

Reply:
When we want to look into the values of variables in Locals and Watch windows etc for another hardware thread (and eventually lanes, when this is merged upstream into gdb) we need to be able to switch to another thread. So, this  method shall enable us to do so.

https://www.intel.com/content/www/us/en/docs/distribution-for-gdb/tutorial-debugging-dpcpp-windows/2024-0/debug-an-application-on-a-gpu.html#:~:text=the%20SIMD%20lane.-,Switch,-to%20another%20active

Reply:
In your scenario, can you just use `--thread {threadId} --frame {frameLevel}` as part of the command string? Or do you need to use versions of GDB and/or commands that don't support that.

There are two problems with this approach:
1. Anything else can come along and change the thread
2. `DefineCurrentThread` isn't really meant to switch the thread, but rather it is meant to update the command factory to know what the current thread is after we enter stop state

Reply:
Setting the thread Id (i.e., defining the selected thread in MICommnadFactory) in a stopped state calls -var-create, -stack-list-arguments, -stack-list-frames, -data-evaluate-expression etc with the intended --thread {threadId} --frame {frameLevel} command without having to manually call each one of them. It is intended that the normal process flow resets the thread the when the program is continued.
Maybe the method here needs to be renamed to ViewSelectedThreadInfo or DisplaySelectedThreadInfo.
Do you have a suggestion here?

Reply:
My problem isn't with the name of the method, but rather that it promises something that it will not deliver on, and make break the UI -- anything else in the UI may change the current thread at any moment. So you are temporarily changing the current thread, but there are no guarantees that the other queries you want to execute against a certain thread will work that way, and doing so without taking the `CommandLock` make break the UI. 

So I would get rid of `ViewSelectedThreadInfo`, and do one of:

1. `ExecuteMICommandWithResultsObject` should take a (optional?) thread id (and maybe frame id while you are at it?) and `MICommandFactory.ThreadCmdAsync` (or `ThreadFrameCmdAsync` if there is a `frameLevel` also) should be used if a thread/frame is provided.
-or-
2. If you only need to deal with versions of GDB that support `--thread {threadId} --frame {frameLevel}`, then you don't need to update MIEngine to solve this part of the problem.

Reply:
Removing ViewSelectedThreadInfo as part of the discussion. ExecuteMICommandWithResultsObject takes an argument which can contain threadId, frameLevel (and laneId when needed, for the future).